src/components/fields/edit/LatLongEditor.jsx

```diff
@@ -15,16 +15,73 @@ function getNumberString(n) {
   return n ? n.toString() : '';
 }
 
-function deriveGpsStringsFromValue(value) {
-  const currentLatitude = get(value, '0', null);
-  const currentLongitude = get(value, '1', null);
-
+function calculateInitialState(coordinates) {
   return {
-    latitudeString: getNumberString(currentLatitude),
-    longitudeString: getNumberString(currentLongitude),
+    latitude: getNumberString(get(coordinates, '0', '')),
+    longitude: getNumberString(get(coordinates, '1', '')),
+    mapCoordinates: null,
+    setterType: 'external',
   };
 }
 
+function reducer(state, action) {
+  switch (action.type) {
+    case 'input-latitude':
```

Comment:
would be good to define reducer actions as constants to prevent typos from messing things up. i'm also still not a fan of these switch statements but it's not that big a deal..